src/main/java/net/dv8tion/jda/api/utils/messages/MessageEditBuilder.java

```diff
@@ -0,0 +1,322 @@
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder setAttachments(@Nullable Collection<? extends AttachedFile> attachments)
+    {
+        this.attachments.clear();
+        set |= ATTACHMENTS;
+        if (attachments != null)
+            this.attachments.addAll(attachments);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public List<? extends AttachedFile> getAttachments()
+    {
+        return Collections.unmodifiableList(attachments);
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder replace(boolean isReplace)
+    {
+        this.replace = isReplace;
+        return this;
+    }
+
+    @Override
+    public boolean isReplace()
+    {
+        return replace;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder setContent(@Nullable String content)
+    {
+        super.setContent(content);
+        set |= CONTENT;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder setEmbeds(@Nonnull Collection<? extends MessageEmbed> embeds)
+    {
+        super.setEmbeds(embeds);
+        set |= EMBEDS;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder setComponents(@Nonnull Collection<? extends LayoutComponent> components)
+    {
+        super.setComponents(components);
+        set |= COMPONENTS;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder applyData(@Nonnull MessageEditData data)
+    {
+        Checks.notNull(data, "Data");
+        this.set = data.getFlags();
+        if ((this.set & ALL) == ALL)
+            this.replace = true;
+
+        if (this.isSet(CONTENT))
+            this.setContent(data.getContent());
+        if (this.isSet(EMBEDS))
+            this.setEmbeds(data.getEmbeds());
+        if (this.isSet(COMPONENTS))
+            this.setComponents(data.getComponents());
+        if (this.isSet(ATTACHMENTS))
+            this.setAttachments(data.getAttachments());
+
+        if (this.isSet(MENTIONS))
+        {
+            String[] empty = new String[0];
+            this.allowedMentions.mentionUsers(data.getMentionedUsers().toArray(empty));
+            this.allowedMentions.mentionRoles(data.getMentionedRoles().toArray(empty));
+            this.allowedMentions.allowedMentions(data.getAllowedMentions());
+            this.allowedMentions.mentionRepliedUser(data.isMentionRepliedUser());
+        }
+
+        return this;
+    }
+
+    @Override
+    public boolean isEmpty()
+    {
+        return !replace && set == 0;
+    }
+
+    @Override
+    public boolean isValid()
+    {
+        if (isSet(EMBEDS) && embeds.size() > Message.MAX_EMBED_COUNT)
+            return false;
+        if (isSet(COMPONENTS) && components.size() > Message.MAX_COMPONENT_COUNT)
+            return false;
+        return !isSet(CONTENT) || Helpers.codePointLength(content) <= Message.MAX_CONTENT_LENGTH;
+    }
+
+    private boolean isSet(int flag)
```

Comment:
can we make this package protected to allow it to be used in `MessageCreateBuilder`?

Reply:
What for?